test/helpers/matchers.spec.ts

```diff
@@ -46,10 +46,16 @@ describe('the middleware path matcher', () => {
       '^(?:\\/(_next\\/data\\/[^/]{1,}))?(?:\\/([^/.]{1,}))\\/shows(?:\\/((?!99|88).*))(.json)?[\\/#\\?]?$',
       '^(?:\\/(_next\\/data\\/[^/]{1,}))?\\/shows(?:\\/((?!99|88).*))(.json)?[\\/#\\?]?$',
     ]
-    for (const regex of regexes) {
-      const stripped = stripLookahead(regex)
-      expect(regex).toMatch(/\(\?!/)
-      expect(stripped).not.toMatch(/\(\?!/)
+
+    const expected = [
```

Comment:
I made these tests more explicit @orinokai.

Reply:
I am able to get this test pass with current code that uses current
```
transform(`/${regex}/`
```

It doesn't seem like we have input that would actually validate the problem that we currently have?

Reply:
Yeah I reverted it and it seems fine. even with the double escape. And the middleware demo still fails with a regex error (extra backslash), so still digging into the actual issue.